doc/user/content/ingest-data/upsert-on-disk.md

```diff
@@ -0,0 +1,214 @@
+---------------------------------------------------------------------------
+                 query timestamp: 1692134262972 (2023-08-15 21:17:42.972) +
+           oracle read timestamp: 1692134262972 (2023-08-15 21:17:42.972) +
+ largest not in advance of upper: 1692134264000 (2023-08-15 21:17:44.000) +
+                           upper:[1692134264001 (2023-08-15 21:17:44.001)]+
+                           since:[1692134262972 (2023-08-15 21:17:42.972)]+
+         can respond immediately: true                                    +
+                        timeline: Some(EpochMilliseconds)                 +
+               session wall time: 1692134263145 (2023-08-15 21:17:43.145) +
+                                                                          +
+ source materialize.public.s (u4, storage):                               +
+                   read frontier:[1692134262972 (2023-08-15 21:17:42.972)]+
+                  write frontier:[1692134264001 (2023-08-15 21:17:44.001)]+
+```
+
+When the `query timestamp` and the `write frontier` of the source (at the bottom) are
+equal or within 1 second, the source is caught up.
+
+Additionally, you can run:
+```sql
+SELECT * from src_
+
+##### Determining resource usage
+
+The following query will give you roughly minute-granularity information about how much memory, cpu,
+and disk space your source is using as it hydrates.
+```sql
+SELECT u.process_id, u.cpu_percent, u.memory_percent, u.disk_percent
+  FROM mz_internal.mz_cluster_replica_utilization u
+  JOIN mz_cluster_replicas cr ON cr.id = u.replica_id
+  JOIN mz_clusters c ON c.id = cr.cluster_id
+  WHERE c.name = 'src_cluster';
+```
+
+To view how much real user data we have ingested from your source (with minute granularity), use:
+```sql
+SELECT u.envelope_state_bytes / 1024 / 1024 / 1024 as ingested_working_set_gib FROM mz_sources s
+  JOIN mz_internal.mz_source_statistics u ON s.id = u.id
+  WHERE s.name = 'src_name';
+```
+
+If you have a rough estimate of the working-set size of your source, periodically running this query can give you an estimate
+of how much progress Materialize has made ingesting your source.
+
+
+##### Problems during hydration
+
+You can use:
+```sql
+SELECT s.process_id, s.status, s.reason, s.updated_at
+  FROM mz_internal.mz_cluster_replica_statuses s
+  JOIN mz_cluster_replicas cr ON cr.id = s.replica_id
+  JOIN mz_clusters c ON c.id = cr.cluster_id
+  WHERE c.name = 'materialize_public_s'
+```
+
+to inspect the status of your replica. If you see problems, like an `oom-killed` (out of memory) reason, you may
+need to adjust your replica size for hydration.
+
+
+4. #### Choosing a size and testing rehydration
+
+##### Choosing a size
+
+After hydration is finished, check the final size of your working set using:
+```sql
+SELECT u.envelope_state_bytes / 1024 / 1024 / 1024 as ingested_working_set_gib FROM mz_sources s
+  JOIN mz_internal.mz_source_statistics u ON s.id = u.id
+  WHERE s.name = 'src_name';
```

Comment:
Instead of getting the working set size by sources, we can get that for all the sources in the cluster. So something like the following instead 
`select SUM(u.envelope_state_bytes) / 1024 / 1024 / 1024 from mz_sources s JOIN mz_internal.mz_source_statistics u ON s.id = u.id WHERE s.cluster_id = '<cluster_id>'`